Code/Tools/FBuild/FBuildCore/Graph/ObjectNode.cpp

```diff
@@ -1454,6 +1454,39 @@ bool ObjectNode::BuildArgs( const Job * job, Args & fullArgs, Pass pass, bool us
                 }
             }
         }
+	
+	//remove static analyzer from clang preprocessor
+	if ( pass == PASS_PREPROCESSOR_ONLY )
+	{
+	   if ( isClang )
+	   {
+	      if ( StripTokenWithArg( "--analyze", token, i ) )
+	      {
+		  continue;
+	      }
+
+	      if ( StripTokenWithArg( "-Xanalyzer", token, i ) )
+	      {
+		  continue;
+	      }
+
+	      if ( StripTokenWithArg( "-analyzer-output", token, i ) )
+	      {
+		  continue;
+	      }
+
+	      if ( StripTokenWithArg( "-analyzer-config", token, i ) )
+	      {
+		  continue;
+	      }
+
+	      if ( StripTokenWithArg( "-analyzer-checker", token, i ) )
+	      {
+		  continue;
+	      }
+	   }
+	}
```

Comment:
The --analyze flag has no parameter, all others do. And I will fix the
spacing issue later today.

On 30 Aug 2016 8:32 AM, "ffulin" <notifications@github.com> wrote:

> In Code/Tools/FBuild/FBuildCore/Graph/ObjectNode.cpp
> <https://github.com/fastbuild/fastbuild/pull/137#discussion_r76732665>:
>
> > +	      if ( StripTokenWithArg( "-analyzer-output", token, i ) )
> > +	      {
> > +		  continue;
> > +	      }
> > +
> > +	      if ( StripTokenWithArg( "-analyzer-config", token, i ) )
> > +	      {
> > +		  continue;
> > +	      }
> > +
> > +	      if ( StripTokenWithArg( "-analyzer-checker", token, i ) )
> > +	      {
> > +		  continue;
> > +	      }
> > +	   }
> > +	}
>
> It looks like your change mixes tabs and spaces and uses inconsistent tab
> widths. It would simplify integration if you can consistently use 4-space
> indentation with no tab characters (all spaces).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/fastbuild/fastbuild/pull/137/files/5f967144a0bfdf14f1458337e61f0c93d65d40b8#r76732665>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AASJwu-OG4byFIGqeuIb4OivNM7XID4rks5qk8BqgaJpZM4JvOpJ>
> .
>